File: app/src/main/java/amagi82/modularcharactersheetcreator/ui/sheet/modules/StatViewModel.java
package amagi82.modularcharactersheetcreator.ui.sheet.modules;

import android.databinding.ObservableField;

import amagi82.modularcharactersheetcreator.models.modules.Module;
import amagi82.modularcharactersheetcreator.models.modules.Stat;
import amagi82.modularcharactersheetcreator.ui._base.BaseModuleViewModel;

public class StatViewModel extends BaseModuleViewModel {
    public final ObservableField<String> title = new ObservableField<>();
    public final ObservableField<String> textBody = new ObservableField<>();
    public final ObservableField<Stat> stat = new ObservableField<>();

    public StatViewModel(Module module) {
        super(module);
    }

    public void update(Module module){
        title.set(module.title());
        textBody.set(module.textBody());
        stat.set(module.stat());
    }
}

File: app/src/main/java/amagi82/modularcharactersheetcreator/ui/sheet/modules/StatViewModel.kt
package amagi82.modularcharactersheetcreator.ui.sheet.modules

import amagi82.modularcharactersheetcreator.models.modules.Module
import amagi82.modularcharactersheetcreator.models.modules.Stat
import amagi82.modularcharactersheetcreator.ui._base.BaseModuleViewModel
import android.databinding.ObservableField

class StatViewModel(module: Module) : BaseModuleViewModel(module) {
    val title = ObservableField<String>()
    val textBody = ObservableField<String>()
    val stat = ObservableField<Stat>()

    init {
        update(module)
    }

    override fun update(module: Module) {
        title.set(module.title)
        textBody.set(module.textBody)
        stat.set(module.stat)
    }
}
